mm2src/common/executor/abortable_system/abortable_queue.rs

```diff
@@ -1,6 +1,7 @@
 use crate::executor::abortable_system::{AbortableSystem, InnerShared, InnerWeak, SystemInner};
 use crate::executor::spawner::{SpawnAbortable, SpawnFuture};
 use crate::executor::{spawn, AbortSettings, Timer};
+use crossbeam::queue::SegQueue;
```

Comment:
`SeqQueue` is definitely not the best solution, but in comparison with `Arc<PaMutex<Vec<FutureId>>>`, `SeqQueue` is not that bad.
I performed a few tests with a fixed size `ArrayQueue` and `PaMutex<Vec<FutureId>>`, and here the results:
1) ITEMS_NUMBER = 5_000_000:
> push_values_lock_free_dynamic took 337.420958ms: items=5000000
pop_values_lock_free_dynamic took 371.537791ms: items=5000000, missing=732
====
vvvv
pop_values_lock_free_array took 328.117666ms: items=5000000, missing=15717
push_values_lock_free_array took 328.12ms: items=5000000
====
vvvv
push_values_mutex took 3.156110208s: items=5000000
pop_values_mutex took 3.154702583s: items=5000000, missing=24710287
====

2) ITEMS_NUMBER = 500_000:
> push_values_lock_free_dynamic took 37.359958ms: items=500000
pop_values_lock_free_dynamic took 41.183875ms: items=500000, missing=0
====
push_values_lock_free_array took 37.030666ms: items=500000
pop_values_lock_free_array took 37.027375ms: items=500000, missing=4524
====
push_values_mutex took 368.443375ms: items=500000
pop_values_mutex took 369.029791ms: items=500000, missing=3175914

The code is available [here](https://pastebin.ubuntu.com/p/Sd8xQPfTJ9/).

Unfortunately, we can't approximate total number of futures, but the difference between `ArrayQueue` and `SegQueue` is not so high to use an approximation.
Also I agree with you that it's not an optimal solution because when we spawn a future, we need to lock the `PaMutex<Inner>`, mutex and then try to pop a `FutureId` from the `Inner::finished_futures` lock-free collection.

But I couldn't find a better solution to avoid using the second thread-synchronization primitive.

Reply:
@ozkanonur what do you think about it?

Reply:
Fixed by https://github.com/KomodoPlatform/atomicDEX-API/pull/1528#issuecomment-1307743512

Reply:
> Fixed by [#1528 (comment)](https://github.com/KomodoPlatform/atomicDEX-API/pull/1528#issuecomment-1307743512)

Awesome :)